openquake/hazardlib/gsim/gsim_table.py

```diff
@@ -0,0 +1,545 @@
+                    numpy.ones_like(dists))
+        return mean_amp, sigma_amps
+
+    def get_mean_table(self, imt, rctx):
+        """
+        Returns amplification factors for the mean, given the rupture and
+        intensity measure type.
+
+        :returns:
+            amplification table as an array of [Number Distances,
+            Number Levels]
+        """
+        # Levels by Distances
+        if isinstance(imt, (imt_module.PGA, imt_module.PGV)):
+            interpolator = interp1d(self.magnitudes,
+                                    numpy.log10(self.mean[str(imt)]), axis=2)
+            output_table = 10.0 ** (
+                interpolator(rctx.mag).reshape(self.shape[0], self.shape[3])
+                )
+        else:
+            # For spectral accelerations - need two step process
+            # Interpolate period - log-log space
+            interpolator = interp1d(numpy.log10(self.periods),
+                                    numpy.log10(self.mean["SA"]),
+                                    axis=1)
+            period_table = interpolator(numpy.log10(imt.period))
+            # Interpolate magnitude - linear-log space
+            mag_interpolator = interp1d(self.magnitudes, period_table, axis=1)
+            output_table = 10.0 ** mag_interpolator(rctx.mag)
+        return output_table
+
+    def get_sigma_tables(self, imt, rctx, stddev_types):
+        """
+        Returns modification factors for the standard deviations, given the
+        rupture and intensity measure type.
+
+        :returns:
+            List of standard deviation modification tables, each as an array
+            of [Number Distances, Number Levels]
+
+        """
+        output_tables = []
+        for stddev_type in stddev_types:
+            # For PGA and PGV only needs to apply magnitude interpolation
+            if isinstance(imt, (imt_module.PGA, imt_module.PGV)):
+                interpolator = interp1d(self.magnitudes,
+                                        self.sigma[stddev_type][str(imt)],
+                                        axis=2)
+                output_tables.append(
+                    interpolator(rctx.mag).reshape(self.shape[0],
+                                                   self.shape[3]))
+
+            else:
+                # For spectral accelerations - need two step process
+                # Interpolate period
+                interpolator = interp1d(numpy.log10(self.periods),
+                                        self.sigma[stddev_type]["SA"],
+                                        axis=1)
+                period_table = interpolator(numpy.log10(imt.period))
+                mag_interpolator = interp1d(self.magnitudes,
+                                            period_table,
+                                            axis=1)
+                output_tables.append(mag_interpolator(rctx.mag))
+        return output_tables
+
+
+class GMPETable(GMPE):
```

Comment:
There was a sort of intention behind that to allow the user to specify a "permanent" GSIM table. That is to create a class that inherits the GMPETable and add the path to the hdf5 file as a property of the GMPE, rather than requiring input at instantiation.